kratos/solving_strategies/schemes/residual_based_bdf_scheme.h

```diff
@@ -386,13 +386,12 @@ class ResidualBasedBDFScheme
         // Getting first node iterator
         const auto it_node_begin = rModelPart.Nodes().begin();
 
-        #pragma omp parallel for
-        for(int i = 0;  i< num_nodes; ++i) {
-            auto it_node = it_node_begin + i;
+        IndexPartition<std::size_t>(num_nodes).for_each([&](std::size_t Index){
+            auto it_node = it_node_begin + Index;
 
             UpdateFirstDerivative(it_node);
             UpdateSecondDerivative(it_node);
```

Comment:
not for this PR but this method should accept a reference and not an iterator
=> then we can use block_for_each

Reply:
Agree. Initially I changed the signature of these but the changes rapidly propagate beyond the core.